src/libsystemd-network/dhcp6-option.c

```diff
@@ -463,7 +463,9 @@ int dhcp6_option_parse_ia(DHCP6Option *iaoption, DHCP6IA *ia) {
                         status = dhcp6_option_parse_status(option, optlen + offsetof(DHCP6Option, data));
                         if (status < 0)
                                 return status;
+
                         if (status > 0) {
+                                *status_code = status;
```

Comment:
hmm, so this is a bit weird. This call now sometimes initializes the status code and sometimes does not, and returns -EINVAL in cases where it initializes it as well as in cases where it does not. 

I think it would make sense to alter the functions return values a bit. Specifically, return 1 if all is good, and initialize status code to 0. Return 0 if we got a status code, and return the status code in that case. And on any other error return the typical negative error code, and do not initialize the value.

As per our coding style we want functions that either succeed and initialize their return parameters or functions that fail and don't touch them, hence we should do the same here too.

Reply:
also, i think we should change the function so that passing NULL as status code pointer is OK too, in case the caller isn't interested.